ompi/mca/coll/accelerator/coll_accelerator_reduce.c

```diff
@@ -70,9 +71,15 @@ mca_coll_accelerator_reduce(const void *sbuf, void *rbuf, size_t count,
         rbuf2 = rbuf; /* save away original buffer */
         rbuf = rbuf1 - gap;
     }
-    rc = s->c_coll.coll_reduce((void *) sbuf, rbuf, count,
-                               dtype, op, root, comm,
-                               s->c_coll.coll_reduce_module);
+
+    if ((comm == NULL) && (root == -1)) {
```

Comment:
This is a horrible highjack of the reduction API ! A collective module is always attached to a communicator (also that's where the `module` is coming from). How can you end up here with a comm equal to NULL ?

Reply:
Don't disagree that the usage is is hacky. Do you have an alternative option that doesn't involve NULL comm and -1 root that also doesn't involve duplicated code? Maybe there are other constants for comm and root that are more appropriate here? 

cc @jsquyres

Reply:
yes, check my other comment. You are trying to do a local_reduce in the reduce collective by using a NULL communicator as a signal. There is an API for local reduce, in this same PR, why don't you use it instead of redirecting it into the reduce ?

Reply:
But that would effectively involve copying the code in `mca_coll_accelerator_reduce` into `mca_coll_accelerator_reduce_local`. I was hoping there was a way to avoid duplicated code.

Reply:
absolutely. Put it into a separate function and invoke it in all places where it is needed.